model_analyzer/perf_analyzer/perf_analyzer.py

```diff
@@ -131,63 +136,78 @@ def run(self, metrics, env=None):
-                            perf_analyzer_env[variable] = os.path.expandvars(
-                                value)
-                try:
-                    process = Popen(cmd,
-                                    start_new_session=True,
-                                    stdout=PIPE,
-                                    stderr=STDOUT,
-                                    encoding='utf-8',
-                                    env=perf_analyzer_env)
-                except FileNotFoundError as e:
-                    raise TritonModelAnalyzerException(
-                        f"perf_analyzer binary not found : {e}")
-
-                if self._poll_perf_analyzer(process) == 1:
-                    # failure
-                    return 1
-
-                if process.returncode > 0:
-                    if self._auto_adjust_parameters(cmd, process) == 1:
-                        return 1
-                elif process.returncode < 0:
-                    logger.error(
-                        'perf_analyzer was terminated by signal: '
-                        f'{signal.Signals(abs(process.returncode)).name}')
-                    return 1
-                else:
-                    self._parse_output(metrics)
+                status = self._execute_pa(env)
+                if status == self.PA_FAIL:
+                    return status
+                elif status == self.PA_SUCCESS:
+                    self._parse_outputs(metrics)
                     break
+                elif status == self.PA_RETRY:
+                    continue
+                else:
+                    raise TritonModelAnalyzerException(
+                        f"Unexpected PA return {status}")
+
             else:
-                if self._config['measurement-mode'] == 'time_windows':
+                if self._base_perf_config['measurement-mode'] == 'time_windows':
                     logger.info(
                         f"Ran perf_analyzer {self._max_retries} times, "
                         "but no valid requests recorded in max time interval"
-                        f" of {self._config['measurement-interval']} ")
-                elif self._config['measurement-mode'] == 'count_windows':
+                        f" of {self._base_perf_config['measurement-interval']} "
+                    )
+                elif self._base_perf_config[
+                        'measurement-mode'] == 'count_windows':
                     logger.info(
                         f"Ran perf_analyzer {self._max_retries} times, "
                         "but no valid requests recorded over max request count"
-                        f" of {self._config['measurement-request-count']} ")
-                return 1
+                        f" of {self._base_perf_config['measurement-request-count']} "
+                    )
+                return self.PA_FAIL
+
+        return self.PA_SUCCESS
+
+    def _execute_pa(self, env):
```

Comment:
Seems like this can be refactored into:

- Set cmd
- Set perf_analyzer_env
- actually execute (try: section)
- poll and check return code

Reply:
done